extensions/amp-date-countdown/1.0/date-countdown.js

```diff
@@ -47,95 +50,104 @@ const TimeUnit = {
 };
 
 // Default prop values
-const DEFAULT_OFFSET_SECONDS = 0;
 const DEFAULT_LOCALE = 'en';
 const DEFAULT_WHEN_ENDED = 'stop';
 const DEFAULT_BIGGEST_UNIT = 'DAYS';
 
+/**
+ * @param {!JsonObject} data
+ * @return {string}
+ */
+const DEFAULT_RENDER = (data) =>
+  /** @type {string} */ (`${data['days']} ${data['dd']}, ` +
+    `${data['hours']} ${data['hh']}, ` +
+    `${data['minutes']} ${data['mm']}, ` +
+    `${data['seconds']} ${data['ss']}, `);
+
 /**
  * @param {!DateCountdownPropsDef} props
  * @return {PreactDef.Renderable}
  */
 export function DateCountdown({
-  endDate,
-  timeleftMs,
-  timestampMs,
-  timestampSeconds,
-  offsetSeconds = DEFAULT_OFFSET_SECONDS,
+  datetime,
   whenEnded = DEFAULT_WHEN_ENDED,
   locale = DEFAULT_LOCALE,
   biggestUnit = DEFAULT_BIGGEST_UNIT,
-  render,
-  children,
+  render = DEFAULT_RENDER,
   ...rest
 }) {
   useResourcesNotify();
   const {playable} = useAmpContext();
-  const epoch = useMemo(
-    () =>
-      getEpoch(endDate, timeleftMs, timestampMs, timestampSeconds) +
-      offsetSeconds * MILLISECONDS_IN_SECOND,
-    [endDate, timeleftMs, timestampMs, timestampSeconds, offsetSeconds]
+
+  const epoch = getDate(datetime);
+
+  // How much time is left in our countdown
+  const [, setTimeleft] = useState(epoch - Date.now());
```

Comment:
I think we can skip some of the ceremony by removing `dataRef`  and `timeLeft` and instead have a `useState` for `rendererData`.

i.e.

```js
const epoch = getDate(datetime);
const localeStrings = useMemo(() => getLocaleWord(locale), [locale]);
const [rendererData, setRendererData] = useState(() => ({
  ... getYDHMSFromMs(...),
  ...localeStrings
});

...

// Now in the setInterval
setRendererData({ ...getYDHMSFromMs(...), ...localData });

// Now back at toplevel
const rendered = useRenderer(render, rendererData);
```

Reply:
Thanks Jake, I used a variation of this in the latest revision.  In `setInterval` I set just `timeleft` and in the main body `timeleft` updates `data` and includes `localeStrings`.

Reply:
Wow! It looks fantastic 🎉